features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/GatewayEnvironments/AddEditGWEnvironment.jsx

```diff
@@ -58,48 +64,73 @@ function reducer(state, { field, value }) {
 }
 
 /**
- * Render a pop-up dialog to add/edit an Gateway label
+ * Render a pop-up dialog to add/edit a Gateway Environment
  * @param {JSON} props .
  * @returns {JSX}.
  */
-function AddEditMGLabel(props) {
+function AddEditGWEnvironment(props) {
     const {
         updateList, dataRow, icon, triggerButtonText, title,
     } = props;
     const classes = useStyles();
 
+    const defaultVhost = {
+        host: '', httpContext: '', httpsPort: 443, httpPort: 80, wssPort: 8099, wsPort: 9099, isNew: true,
+    };
     const [initialState, setInitialState] = useState({
+        displayName: '',
         description: '',
-        hosts: [],
+        vhosts: [defaultVhost],
     });
     const [editMode, setIsEditMode] = useState(false);
 
     const [state, dispatch] = useReducer(reducer, initialState);
-    const { name, description, hosts } = state;
+    const {
+        name, displayName, description, vhosts,
+    } = state;
 
     const onChange = (e) => {
         dispatch({ field: e.target.name, value: e.target.value });
     };
     useEffect(() => {
         setInitialState({
+            displayName: '',
             description: '',
-            hosts: [],
+            vhosts: [defaultVhost],
         });
     }, []);
 
-    const handleHostValidation = (hostName) => {
-        if (hostName === undefined) {
+    const handleHostValidation = (vhost) => {
+        if (vhost === undefined) {
             return false;
         }
-        const schema = Joi.string().uri().empty();
-        const validationError = schema.validate(hostName).error;
+        if (!vhost.host) {
+            return 'Host of Vhost is empty';
+        }
+        // same pattern used in admin Rest API
+        const hostPattern = '^(((\\*[\\.-])?(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*'
```

Comment:
Shall we move this validation patter to [`APIValidation.js`](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/APIValidation.js)

Reply:
will be fixed in https://github.com/wso2/product-apim/issues/10003